### ISSUE TITLE
Qiskit support fails to load

### ISSUE THREAD

**Issue by jwoehr** [QuTech-Delft/quantuminspire#84]
# Bug report

In `master` the phrase `from quantuminspire.qiskit import QI` raises `ModuleNotFoundError: No module named 'qiskit.validation.base'`

## What did you try to do?

import quantuminspire.qiskit.QI
(see https://github.com/jwoehr/qis_job)

## What did you expect to happen?

Load as it had been doing. (This has worked for months prior)

## What actually happened?
```
  File "qis_job/qis_job.pyx", line 31, in init qis_job.qis_job
    from quantuminspire.qiskit import QI
  File "/home/jax/work/QISKit/quantuminspire/src/quantuminspire/qiskit/__init__.py", line 1, in <module>
    from quantuminspire.qiskit.quantum_inspire_provider import QuantumInspireProvider
  File "/home/jax/work/QISKit/quantuminspire/src/quantuminspire/qiskit/quantum_inspire_provider.py", line 26, in <module>
    from quantuminspire.qiskit.backend_qx import QuantumInspireBackend
  File "/home/jax/work/QISKit/quantuminspire/src/quantuminspire/qiskit/backend_qx.py", line 31, in <module>
    from qiskit.validation.base import Obj
ModuleNotFoundError: No module named 'qiskit.validation.base'
```

## Environment

Please specify:
* operating system, version and distribution:
```
 Linux xxxxxx 4.15.0-99-generic #100-Ubuntu SMP Wed Apr 22 20:32:56 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
NAME="Ubuntu"
VERSION="18.04.4 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.4 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```
* python version (`python --version`) and distribution: Python 3.6.9 (apt installed)
* list of installed packages (`pip list` or equivalent):
```
alabaster (0.7.12)
alembic (0.9.3.dev0)
anaconda (0.0.1.1)
antlr4-python3-runtime (4.7.2)
appdirs (1.4.3)
apturl (0.5.2)
argh (0.26.2)
arrow (0.15.5)
asn1crypto (0.24.0)
astroid (2.3.3)
asv (0.4.1)
atomicwrites (1.3.0)
attrs (19.3.0)
autopep8 (1.5.1)
Babel (2.8.0)
backcall (0.1.0)
beautifulsoup4 (4.6.0)
bleach (3.1.4)
blinker (1.4)
Brlapi (0.6.6)
certifi (2020.4.5.1)
cffi (1.14.0)
chardet (3.0.4)
click (6.7)
cliff (2.18.0)
cloudpickle (1.3.0)
cmake (3.16.3)
cmd2 (0.8.9)
colorama (0.3.7)
command-not-found (0.3)
coreapi (2.3.3)
coreschema (0.0.4)
coverage (4.5.4)
cplex (12.9.0.0)
cryptography (2.9)
cupshelpers (1.0)
cvxopt (1.2.3)
cvxpy (1.0.25)
cycler (0.10.0)
Cython (0.29.13)
ddt (1.2.1)
decorator (4.4.2)
defer (1.0.6)
defusedxml (0.6.0)
diff-match-patch (20181111)
dill (0.3.1.1)
discover (0.4.0)
distro-info (0.18ubuntu0.18.04.1)
dlx (1.0.4)
docloud (1.0.375)
docplex (2.10.155)
docutils (0.16)
dulwich (0.19.13)
ecos (2.0.7.post1)
entrypoints (0.3)
extras (1.0.0)
fastdtw (0.3.2)
fastjsonschema (2.14.3)
feedparser (5.2.1)
filelock (3.0.12)
fixtures (3.0.0)
flake8 (3.7.9)
Flask (0.12.2)
Flask-BabelEx (0.9.3)
Flask-Compress (1.4.0)
Flask-Gravatar (0.4.2)
Flask-Login (0.4.0)
Flask-Mail (0.9.1)
Flask-Migrate (2.1.1)
Flask-Paranoid (0.2.0)
Flask-Principal (0.4.0)
Flask-Security (1.7.5)
Flask-SocketIO (4.2.1)
Flask-SQLAlchemy (2.1)
Flask-WTF (0.14.2)
future (0.18.2)
h5py (2.10.0)
helpdev (0.6.10)
html5lib (0.999999999)
httplib2 (0.9.2)
hypothesis (4.41.3)
idna (2.9)
imagesize (1.2.0)
immutables (0.6)
importlib-metadata (1.6.0)
inflection (0.3.1)
intervaltree (3.0.2)
ipykernel (5.2.1)
ipython (7.13.0)
ipython-genutils (0.2.0)
ipywidgets (7.5.1)
isort (4.3.21)
itsdangerous (0.24)
itypes (1.1.0)
jedi (0.15.2)
jeepney (0.4.3)
Jinja2 (2.11.2)
joblib (0.14.0)
jsonschema (3.2.0)
jupyter (1.0.0)
jupyter-client (6.1.3)
jupyter-console (6.0.0)
jupyter-core (4.6.3)
jupyter-sphinx (0.2.3)
keyring (21.2.0)
keyrings.alt (3.4.0)
kiwisolver (1.1.0)
language-selector (0.1)
launchpadlib (1.10.6)
lazr.restfulclient (0.13.5)
lazr.uri (1.0.3)
lazy-object-proxy (1.4.3)
linecache2 (1.0.0)
louis (3.5.0)
lxml (4.2.1)
macaroonbakery (1.1.3)
Mako (1.0.7)
MarkupSafe (1.1.1)
marshmallow (3.2.1)
marshmallow-polyfield (5.7)
matplotlib (3.1.1)
mccabe (0.6.1)
memory-profiler (0.57.0)
mistune (0.8.4)
more-itertools (5.0.0)
mpmath (1.1.0)
msgpack (0.6.2)
multiprocess (0.70.9)
nbconvert (5.6.1)
nbformat (5.0.5)
nbimporter (0.3.1)
ndg-httpsclient (0.5.1)
nest-asyncio (1.0.0)
netifaces (0.10.4)
networkx (2.4)
nlopt (2.6.1)
nose (1.3.7)
notebook (6.0.1)
ntlm-auth (1.4.0)
numpy (1.18.1)
numpydoc (0.9.2)
nuqasm2 (0.33, /home/jax/work/QISKit/nuqasm2)
nxpd (0.2.0)
oauth (1.0.1)
oauthlib (2.0.6)
olefile (0.45.1)
osqp (0.6.1)
packaging (20.3)
paho-mqtt (1.5.0)
pandas (0.25.1)
pandocfilters (1.4.2)
parameterized (0.7.0)
paramiko (2.0.0)
parso (0.5.2)
passlib (1.7.1)
pathtools (0.1.2)
pbr (5.4.3)
pexpect (4.8.0)
pickleshare (0.7.5)
Pillow (5.1.0)
pip (9.0.1)
pkg-resources (0.0.0)
pkginfo (1.5.0.1)
pluggy (0.13.1)
ply (3.11)
prettytable (0.7.2)
prometheus-client (0.7.1)
prompt-toolkit (2.0.10)
protobuf (3.0.0)
psutil (5.7.0)
psycopg2 (2.8.5)
ptyprocess (0.6.0)
py (1.8.0)
pyasn1 (0.4.2)
pybind11 (2.4.3)
pycairo (1.16.2)
pycodestyle (2.5.0)
pycparser (2.20)
pycrypto (2.6.1)
pycups (1.9.73)
pydocstyle (5.0.2)
pydot (1.4.1)
pyenchant (2.0.0)
pyflakes (2.2.0)
Pygments (2.6.1)
pygobject (3.26.1)
pyinotify (0.9.6)
PyJWT (1.5.3)
pylatexenc (2.1)
pylint (2.4.4)
pylintfileheader (0.0.2)
pymacaroons (0.13.0)
Pympler (0.8)
PyNaCl (1.1.2)
pyodbc (4.0.30)
pyopencl (2019.1.1)
pyOpenSSL (17.5.0)
pyparsing (2.4.7)
pyperclip (1.7.0)
PyQt5 (5.12.3)
PyQt5-sip (12.7.2)
PyQtWebEngine (5.12.1)
pyquante2 (0.1, /home/jax/work/QISKit/pyquante2)
pyquil (2.16.0, /home/jax/work/Rigetti/pyquil)
pyRFC3339 (1.0)
pyrsistent (0.16.0)
pyscf (1.6.4)
pytest (5.2.0)
python-apt (1.6.5+ubuntu0.2)
python-constraint (1.4.0)
python-dateutil (2.8.1)
python-debian (0.1.32)
python-engineio (3.11.2)
python-jsonrpc-server (0.3.4)
python-language-server (0.31.9)
python-mimeparse (1.6.0)
python-rapidjson (0.9.1)
python-socketio (4.4.0)
python-subunit (1.3.0)
pytools (2019.1.1)
pytz (2019.3)
pyxdg (0.26)
PyYAML (5.2)
pyzmq (19.0.0)
qcgpu (0.1.1)
QDarkStyle (2.8.1)
qis-job (3.3)
qiskit (0.14.0, /home/jax/work/QISKit/DEV/qiskit)
qiskit-aer (0.6.0, /home/jax/work/QISKit/DEV/qiskit-aer)
qiskit-aqua (0.8.0, /home/jax/work/QISKit/qiskit_dev_venv4/lib/python3.6/site-packages/qiskit_aqua-0.8.0-py3.6.egg)
qiskit-aqua-interfaces (0.2.1, /home/jax/work/QISKit/qiskit_dev_venv4/lib/python3.6/site-packages/qiskit_aqua_interfaces-0.2.1-py3.6.egg)
qiskit-ibmq-provider (0.7.0, /home/jax/work/QISKit/DEV/qiskit-ibmq-provider)
qiskit-ignis (0.4.0, /home/jax/work/QISKit/qiskit_dev_venv4/lib/python3.6/site-packages/qiskit_ignis-0.4.0-py3.6.egg)
qiskit-jku-provider (1.0.0, /home/jax/work/QISKit/qiskit_dev_venv4/lib/python3.6/site-packages/qiskit_jku_provider-1.0.0-py3.6-linux-x86_64.egg)
qiskit-terra (0.15.0, /home/jax/work/QISKit/qiskit_dev_venv4/lib/python3.6/site-packages/qiskit_terra-0.15.0-py3.6-linux-x86_64.egg)
qrcode (6.1)
QtAwesome (0.7.1)
qtconsole (4.7.3)
QtPy (1.9.0)
Quandl (3.4.8)
quantastica-qconvert (0.9.12)
quantastica-qiskit-forest (0.9.13, /home/jax/work/QISKit/qiskit-forest)
quantuminspire (1.2.0, /home/jax/work/QISKit/quantuminspire/src)
qutip (4.5.0)
readme-renderer (24.0)
reno (2.11.3)
reportlab (3.4.0)
requests (2.23.0)
requests-ntlm (1.1.0)
requests-toolbelt (0.9.1)
requests-unixsocket (0.1.5)
retworkx (0.3.2)
rope (0.16.0)
rpcq (3.0.0)
ruamel.yaml (0.16.5)
ruamel.yaml.clib (0.2.0)
scikit-build (0.10.0)
scikit-learn (0.21.3)
scipy (1.3.1)
screen-resolution-extra (0.0.0)
scs (2.1.1.post2)
seaborn (0.9.0)
SecretStorage (3.1.2)
Send2Trash (1.5.0)
setuptools (46.1.3)
simplejson (3.13.2)
six (1.14.0)
sklearn (0.0)
snowballstemmer (2.0.0)
sortedcontainers (2.1.0)
Sphinx (3.0.1)
sphinx-automodapi (0.12)
sphinx-rtd-theme (0.4.3)
sphinx-tabs (1.1.13)
sphinxcontrib-applehelp (1.0.2)
sphinxcontrib-devhelp (1.0.2)
sphinxcontrib-htmlhelp (1.0.3)
sphinxcontrib-jsmath (1.0.1)
sphinxcontrib-qthelp (1.0.3)
sphinxcontrib-serializinghtml (1.1.4)
sphinxcontrib-websupport (1.1.0)
spyder (4.1.2)
spyder-kernels (1.9.0)
SQLAlchemy (1.1.11)
sqlparse (0.2.4)
ssh-import-id (5.7)
sshtunnel (0.1.4)
stestr (2.6.0)
stevedore (1.31.0)
sympy (1.4)
system-service (0.3)
systemd-python (234)
terminado (0.8.2)
testpath (0.4.4)
testtools (2.3.0)
toml (0.10.0)
torch (1.3.0)
tornado (6.0.4)
tox (3.14.0)
tqdm (4.31.1)
traceback2 (1.4.0)
traitlets (4.3.3)
twine (1.13.0)
typed-ast (1.4.1)
ubuntu-drivers-common (0.0.0)
ufw (0.36)
ujson (1.35)
unattended-upgrades (0.1)
unittest2 (1.1.0)
unity-scope-calculator (0.1)
unity-scope-chromiumbookmarks (0.1)
unity-scope-colourlovers (0.1)
unity-scope-devhelp (0.1)
unity-scope-firefoxbookmarks (0.1)
unity-scope-manpages (0.1)
unity-scope-openclipart (0.1)
unity-scope-texdoc (0.1)
unity-scope-tomboy (0.1)
unity-scope-virtualbox (0.1)
unity-scope-yelp (0.1)
unity-scope-zotero (0.1)
uritemplate (3.0.0)
urllib3 (1.25.9)
usb-creator (0.3.3)
virtualenv (16.7.7)
voluptuous (0.11.7)
wadllib (1.3.2)
watchdog (0.10.2)
wcwidth (0.1.9)
webencodings (0.5.1)
websockets (7.0)
Werkzeug (0.16.0)
wheel (0.30.0)
widgetsnbextension (3.5.1)
wrapt (1.11.2)
WTForms (2.1)
wurlitzer (2.0.0)
xkit (0.0.0)
yapf (0.29.0)
z3-solver (4.8.7.0)
zipp (3.1.0)
zope.interface (4.3.2)
```
* anything else that might be relevant
n/a

**Comment by QFer** [comment]
Hi @jwoehr, qiskit terra is currently at version 0.14.1. (installed with latest qiskit version 0.19.1). I see you have version 0.15.0 installed. This is causing your problem because it doesn't contain qiskit.validation.base anymore. You are a little bit ahead of us.
We also cannot release a fix because it would break current setups for users who do install qiskit from pypi (and getting terra 0.14.1), so we have to wait before qiskit terra 0.15.0 is released as part of qiskit.
I label this one as enhancement because we can check of we can solve it and be backwards compatible with earlier versions of terra.

**Comment by jwoehr** [comment]
Thank you.

**Comment by andrenmelo** [comment]
Hi @QFer, just a heads up that the latest version of qiskit (0.20.0) is shipped with qiskit-terra 0.15.1, so this should now be fixable.

**Comment by jwoehr** [comment]
Now works for me
